indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,200 @@
-import {usePermissionInfo} from 'indico/react/components/principals/hooks';
-import EquipmentList from './EquipmentList';
-import DailyAvailability from './DailyAvailability';
-import NonBookablePeriods from './NonBookablePeriods';
-import RoomPhoto from './RoomPhoto';
-import {actions as roomsActions} from '../../common/rooms';
-import {actions as userActions} from '../../common/user';
-import * as roomsSelectors from './selectors';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
 
 import './RoomEditModal.module.scss';
-
-function isInvalidNotificationPeriod(days) {
-  return days !== null && (days > 30 || days < 1);
-}
-
-function validate(fields) {
-  const {
-    building,
-    floor,
-    number,
-    capacity,
-    surfaceArea,
-    maxAdvanceDays,
-    bookingLimitDays,
-    nonbookablePeriods,
-    owner,
-    latitude,
-    longitude,
-  } = fields;
-  const errors = {};
-  if (!building) {
-    errors.building = Translate.string('Please provide a building.');
-  }
-  if (!floor) {
-    errors.floor = Translate.string('Please provide a floor.');
-  }
-  if (!number) {
-    errors.number = Translate.string('Please provide a number.');
-  }
-  if (capacity < 1) {
-    errors.capacity = Translate.string('Please provide a valid capacity number.');
-  }
-  if (surfaceArea && surfaceArea < 1) {
-    errors.surfaceArea = Translate.string('Please provide a valid surface area.');
-  }
-  if (maxAdvanceDays !== null && maxAdvanceDays < 1) {
-    errors.maxAdvanceDays = Translate.string(
-      'The max. advance booking time must be at least 1 day or empty.'
-    );
-  }
-  if (bookingLimitDays !== null && bookingLimitDays < 1) {
-    errors.bookingLimitDays = Translate.string(
-      'The max. booking duration must be at least 1 day or empty.'
-    );
-  }
-  const notificationPeriodFields = [
-    'notificationBeforeDays',
-    'notificationBeforeDaysWeekly',
-    'notificationBeforeDaysMonthly',
-    'notificationBeforeEndDaily',
-    'notificationBeforeEndWeekly',
-    'notificationBeforeEndMonthly',
-  ];
-
-  notificationPeriodFields.forEach(field => {
-    const value = fields[field];
-    if (isInvalidNotificationPeriod(value)) {
-      errors[field] = Translate.string('Number of days must be between 1 and 30');
-    }
+import {getChangedValues, handleSubmitError, parsers as p} from 'indico/react/forms';
+import RoomPhoto from './RoomPhoto';
+import RoomEditNotifications from './RoomEditNotifications';
+import RoomEditLocation from './RoomEditLocation';
+import RoomEditDetails from './RoomEditDetails';
+import RoomEditPermissions from './RoomEditPermissions';
+import RoomEditOptions from './RoomEditOptions';
+
+function RoomEditModal({roomId, locationId, onClose, afterCreation}) {
+  const [globalAttributes, setGlobalAttributes] = useState([]);
+  const [newRoomId, setNewRoomId] = useState(null);
+  const [wasEverUpdated, setWasEverUpdated] = useState(null);
+
+  const [roomDetails, setRoomDetails] = useState({
```

Comment:
Do we need this state? Ideally it should be enough to pass all the data into the FinalForm's `initialValues`.

Reply:
Do you mean the state inside `roomDetails`? I copied that from the previous default room state set when a new room is created, instead of edited. I believe those are the default values. Not sure if any of them are needed, but I'll give it a try.

Reply:
I updated this part to only contain the essential. Should be fine now.